src/librbd/DiffIterate.cc

```diff
@@ -264,7 +264,11 @@ int DiffIterate::execute() {
     RWLock::RLocker l(m_image_ctx.snap_lock);
     RWLock::RLocker l2(m_image_ctx.parent_lock);
     uint64_t overlap = end_size;
-    m_image_ctx.get_parent_overlap(from_snap_id, &overlap);
+    if (m_image_ctx.snaps.empty()) {
+      m_image_ctx.get_parent_overlap(CEPH_NOSNAP, &overlap);
+    } else {
+      m_image_ctx.get_parent_overlap(m_image_ctx.snaps.back(), &overlap);
```

Comment:
Whoops -- sorry, thinking about this again, I think you can just use ```m_image_ctx.snap_id``` in both cases because that acts as the pseudo-"to_snap_id". When you are comparing to HEAD (or don't have any snapshots), it will be set to CEPH_NOSNAP; otherwise, it will retrieve the minimal overlap between the parent and clone (since the overlap can only shrink).